src/sentry/models/project.py

```diff
@@ -324,6 +327,19 @@ def transfer_to(self, team=None, organization=None):
         if team is not None:
             self.add_team(team)
 
+        # Remove alert owners not in new org
+        alert_rules = AlertRule.objects.fetch_for_project(self).filter(owner_id__isnull=False)
+        rules = Rule.objects.filter(owner_id__isnull=False, project=self)
+        for rule in list(chain(alert_rules, rules)):
+            actor = rule.owner
+            if actor.type == ACTOR_TYPES["user"]:
+                is_member = organization.member_set.filter(user=actor.resolve()).exists()
+            if actor.type == ACTOR_TYPES["team"]:
+                is_member = actor.resolve().organization.id == organization.id
+            if not is_member:
+                rule.owner = None
+                rule.save()
```

Comment:
You can use `rule.update(owner=None)`. Is a little safer since we just update the one field